models/inherit_stock.py

```diff
@@ -15,31 +12,61 @@ class stock_warehouse(models.Model):
     _inherit = "stock.warehouse"
     _inherits = {"eyekraft.shop": 'shop_id'}
 
-    # compute latitude & longitude by Yandex service
     def _get_lat_lng(self):
-	if not self.full_address:
-	    return 0,0
-        if 'geopy' in sys.modules:
-            geolocator = geopy.geocoders.Yandex()
-            for i in xrange(10):
-                try:
-                    location = geolocator.geocode(self.full_address, timeout=10)
-                    break
-                except geopy.exc.GeocoderTimedOut as e:
-                    continue
-                except geopy.exc.GeocoderQueryError as e:
-                    continue
-                except AttributeError:
-                    continue
-                except geopy.exc.GeocoderParseError as e:
-                    continue
-            if not location:
-                _logger.warning("Yandex is unreachable, therefore lat|lng for shop is not computed")
-        else:
-            _logger.warning("Geopy has not been imported, therefore lat|lng for shop is not computed")
-	return location.latitude, location.longitude
-
-    @api.multi
+        """ Compute latitude & longitude by Yandex service
+            :returns: location.latitude, location.longitude
+        """
+
+        # Yandex API Key
+        YANDEX_API_KEY = self.env['ir.config_parameter'].sudo().get_param('web_yandex_maps.api_key')
+
+        if not self.full_address:
+            return 0,0
+            if 'geopy' in sys.modules:
+                # Geocoder using the Yandex Maps API.
+                geolocator = geocoders.Yandex(api_key=YANDEX_API_KEY)
+
+                # https://geopy.readthedocs.io/en/latest/index.html#geopy.geocoders.Yandex.geocode
+                spanish = ["es_ES", "es_AR", "es_BO", "es_CL", "es_CO", "es_CR", "es_DO", "es_EC", "es_GT", "es_MX", "es_PA", "es_PE", "es_PY", "es_UY", "es_VE"]
+                english = ["en_AU", "en_CA", "en_GB", "en_US"]
+                if self.env.context['lang'] == u'es' or self.env.context['lang'] in spanish:
+                    language="en_US"
+                elif self.env.context['lang'] == u'en' or self.env.context['lang'] in english:
+                    language="en_US"
+                elif self.env.context['lang'] == u'tr' or self.env.context['lang'] == u'tr_TR':
+                    language="tr_TR"
+                elif self.env.context['lang'] == u'uk' or self.env.context['lang'] == u'uk_UA':
+                    language="uk_UA"
+                elif self.env.context['lang'] == u'be' or self.env.context['lang'] == u'be_BY':
+                    language="be_BY"
+                elif self.env.context['lang'] == u'ru' or self.env.context['lang'] == u'ru_RU':
+                    language="ru_RU"
+                else:
+                    language="en_RU"
+
+                for i in xrange(10):
```

Comment:
@Tatider If I understand: The structure is always IF then ELIF and finally ELSE (which is executed in case none of the previous IF/ELIF is fulfilled), what do you think?